src/interpolations.jl

```diff
@@ -1774,3 +1783,299 @@ function mapping_type end
 
 mapping_type(::ScalarInterpolation) = IdentityMapping()
 mapping_type(::VectorizedInterpolation) = IdentityMapping()
+
+
+#####################################
+# RaviartThomas (1st kind), H(div)       #
+#####################################
+# https://defelement.com/elements/raviart-thomas.html
+# https://defelement.com/elements/qdiv.html
+struct RaviartThomas{vdim, shape, order} <: VectorInterpolation{vdim, shape, order} end
+mapping_type(::RaviartThomas) = ContravariantPiolaMapping()
+n_dbc_components(::RaviartThomas) = 1
+reference_coordinates(ip::RaviartThomas{vdim}) where {vdim} = fill(NaN * zero(Vec{vdim}), getnbasefunctions(ip))
+dirichlet_edgedof_indices(ip::RaviartThomas{2}) = edgedof_interior_indices(ip)
+dirichlet_facedof_indices(ip::RaviartThomas{3}) = facedof_interior_indices(ip)
+
+# RefTriangle, 1st order Lagrange
+# https://defelement.com/elements/examples/triangle-raviart-thomas-lagrange-1.html
+# Signs changed when needed to make positive direction outwards
+function reference_shape_value(ip::RaviartThomas{2, RefTriangle, 1}, ξ::Vec{2}, i::Int)
+    x, y = ξ
+    i == 1 && return ξ                  # Flip sign
+    i == 2 && return Vec(x - 1, y)        # Keep sign
+    i == 3 && return Vec(x, y - 1)      # Flip sign
+    throw(ArgumentError("no shape function $i for interpolation $ip"))
+end
+
+getnbasefunctions(::RaviartThomas{2, RefTriangle, 1}) = 3
+edgedof_interior_indices(::RaviartThomas{2, RefTriangle, 1}) = ((1,), (2,), (3,))
+adjust_dofs_during_distribution(::RaviartThomas) = false
+
+function get_direction(::RaviartThomas{2, RefTriangle, 1}, j, cell)
+    edge = edges(cell)[j]
+    return ifelse(edge[2] > edge[1], 1, -1)
+end
+
+# RefTriangle, 2st order Lagrange
+#=
+----------------+--------------------
+Vertex numbers: | Vertex coordinates:
+    2           |
+    | \         | v1: 𝛏 = (1.0, 0.0)
+    |   \       | v2: 𝛏 = (0.0, 1.0)
+ξ₂^ |     \     | v3: 𝛏 = (0.0, 0.0)
+  | 3-------1   |
+  +--> ξ₁       |
+----------------+--------------------
+Edge numbers:   | Edge identifiers:
+    +           |
+    | \         | e1: (v1, v2)
+    2   1       | e2: (v2, v3)
+    |     \     | e3: (v3, v1)
+    +---3---+   |
+----------------+--------------------
+```
+"""
+RefTriangle
+=#
+# https://defelement.com/elements/examples/triangle-raviart-thomas-lagrange-2.html
+# Signs changed when needed to make positive direction outwards
+function reference_shape_value(ip::RaviartThomas{2, RefTriangle, 2}, ξ::Vec{2}, i::Int)
+    x, y = ξ
+    # Face 1 (keep ordering, flip sign)
+    i == 1 && return Vec(4x * (2x - 1), 2y * (4x - 1)) / 2
```

Comment:
The division by 2 compared to defelement gives an identity for the line integral of 1/num_edge_dofs, such that setting the actual dof-values to 1 results in a unit flux. @termi-official: Do you see any issues by scaling the shape functions like this?

Reply:
The advantage is that this way, we don't have to do strange modifications to the values when applying boundary conditions, see the tests at the end of test_interpolations.jl